x/events/events.go

```diff
@@ -0,0 +1,49 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package events
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/ory/x/otelx/semconv"
+)
+
+const (
+	// LoginAccepted will be emitted when the login UI accepts a login request.
+	LoginAccepted semconv.Event = "OAuth2LoginAccepted"
+
+	// LoginRejected will be emitted when the login UI rejects a login request.
+	LoginRejected semconv.Event = "OAuth2LoginRejected"
+
+	// ConsentGranted will be emitted when the consent UI accepts a consent request.
+	ConsentGranted semconv.Event = "OAuth2ConsentGranted"
+
+	// ConsentRejected will be emitted when the consent UI rejects a consent request.
+	ConsentRejected semconv.Event = "OAuth2ConsentRejected"
+
+	// ConsentRevoked will be emitted when the user revokes a consent request.
+	ConsentRevoked semconv.Event = "OAuth2ConsentRevoked"
+
+	// AccessTokenCreated will be emitted by requests to POST /oauth2/token in case the request was successful.
+	AccessTokenCreated semconv.Event = "OAuth2AccessTokenCreated" //nolint:gosec
+
+	// AccessTokenNotCreated will be emitted by requests to POST /oauth2/token in case the request was unsuccessful.
+	AccessTokenNotCreated semconv.Event = "OAuth2AccessTokenNotCreated" //nolint:gosec
```

Comment:
This feels weird. Do we really need this?

Reply:
It was in the list of needed events, so I guess? @kmherrmann ?

Reply:
Maybe make it more generic and call it OAuh2FlowError ?

Reply:
We probably don't need this event at all. What it's trying to signal is some error during the OAuth2 flow. But many many things can go wrong during the flow, and we're not about to signal all of them to our analytics platform, are we?

Errors typically don't get emitted as analytics events but go to the monitoring platform (Grafana) as logs or error traces.

Reply:
Good points Arne! I think we can still ship this for now and invest some time in better observability next. We don't have to use the events and if they serve no purpose in Analytics, we'll remove them